front/app/modules/commercial/insights/hooks/useInsightsInputs.test.ts

```diff
@@ -0,0 +1,152 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import useInsightsInputs, { QueryParameters } from './useInsightsInputs';
+import { Observable, Subscription } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import { waitFor } from 'utils/testUtils/rtl';
+import { insightsInputsStream } from 'modules/commercial/insights/services/insightsInputs';
+
+const viewId = '1';
+
+const queryParameters: QueryParameters = {
+  category: '3',
+  pageSize: 10,
+  pageNumber: 12,
+  search: 'search',
+  sort: '-approval',
+};
+
+const mockInputs = {
+  data: {
+    currentPage: 1,
+    lastPage: 2,
```

Comment:
I think here we'd like to mock the data as it's returned by the back-end (and the stream), so with a type of `IInsightsInputs`, with links instead of currentPage and lastPage. But then I get confused how this works, so I might not have completely understood if you're mocking the observable or the subscription ?

Reply:
You are right, of course! The data should be as it comes from the service - we are testing that the hook transforms it correctly (among other things). This should be fixed now!